src/index.ts

```diff
@@ -74,39 +97,32 @@ export interface MagicLinkGenerationOptions {
 }
 
 /**
- * The store TOTP configuration.
+ * The create TOTP CRUD method.
+ * @param data The TOTP data.
+ * @param expiresAt The TOTP expiration date.
  */
-export interface StoreTOTPOptions {
-  /**
-   * The encrypted TOTP.
-   */
-  hash: string
-
-  /**
-   * The status of the TOTP.
-   * @default true
-   */
-  active: boolean
-
-  /**
-   * The number of attempts the user tried to verify the TOTP.
-   * @default 0
-   */
-  attempts: number
+export interface CreateTOTP {
+  (data: TOTPData, expiresAt: Date): Promise<void>
```

Comment:
Date should be suitable since it includes time. Note that an ORM like Prisma has way to map Date's to database types and vice versa.

https://www.prisma.io/docs/orm/reference/prisma-schema-reference#datetimeDateTime

Remarks
Prisma Client returns all DateTime as native Date objects.

Connector	Default mapping
PostgreSQL	timestamp(3)
SQL Server	datetime2
MySQL	        DATETIME(3)
MongoDB	Timestamp
SQLite	         NUMERIC
CockroachDB	TIMESTAMP